### DISCUSSION TITLE
Indexed. Use of moved value

### DISCUSSION THREAD

**Discussion by ghost** [sycamore-rs/sycamore#515] [Q&A] [answered]
Please tell me what is the best way to write something like this so that there is no error?

_(use of moved value: `value.foo`
move occurs because `value.foo` has type `std::string::String`, which does not implement the `Copy` traitrustc[E0382](https://doc.rust-lang.org/error-index.html#E0382))_

```
use sycamore::prelude::*;

#[derive(Clone, PartialEq)]
struct Foo {
    foo: String,
}

#[component]
fn Test<G: Html>(cx: Scope) -> View<G> {
    let values = create_signal(cx, vec![] as Vec<Foo>);

    view! { cx,
            Indexed(
                iterable=values,
                view=|cx, value| view! { cx,
                    div(data-value=value.foo) { (value.foo) }
                }
            )
    }
}
```

**Comment by ghost** [comment, marked as answer]
Is it ok to use `create_ref` like this?

```
Indexed(
    iterable=values,
    view=|cx, value| {
        let value = create_ref(cx, value);
        view! { cx,
            div(data-value=value.foo) { (value.foo) }
        }
    }
)
```

**Reply by lukechu10** [reply]
Yup that works great.

**Reply by ghost** [reply]
won't there be memory leaks if the vector is frequently overwritten (loaded from the network)?

**Reply by lukechu10** [reply]
> won't there be memory leaks if the vector is frequently overwritten (loaded from the network)?

No because the `cx` refers to the one inside the closure so it's destroyed every time that row changes.

**Comment by nicmr** [comment]
@lukechu10 The accepted solution does not work anymore in sycamore >= 0.9.0-beta2, since create_ref was removed from sycamore_reactive. How would we handle this now?

**Reply by lukechu10** [reply]
The simplest here would just be to clone the string. If that is not acceptable, you can always put it into a `Rc` but that is almost never the case since the overhead of sending the string to JS will outweigh many times any cost of cloning the string.
